src/braket/default_simulator/density_matrix_simulator.py

```diff
@@ -120,9 +213,21 @@ def properties(self) -> GateModelSimulatorDeviceCapabilities:
                             {"name": "Probability", "minShots": 0, "maxShots": max_shots},
                             {"name": "DensityMatrix", "minShots": 0, "maxShots": 0},
                         ],
-                    }
+                    },
                 },
                 "paradigm": {"qubitCount": qubit_count},
                 "deviceParameters": GateModelSimulatorDeviceParameters.schema(),
             }
         )
+
+    @singledispatchmethod
+    def _run_internal(
```

Comment:
Maybe it's just my opinion, but I prefer the default implementation of a `singledispatch` to be either an error or "true" default behavior that doesn't care about the type. In particular, the default implementation should not have a type at all.

Reply:
Ya I vibe with that, I'll change this